File: apps/web/lib/api/gallery.ts
import { getApiBaseUrl } from "./config";

/** M8d — public gallery card (no owner / draft / code). */
export type GalleryCard = {
  publicId: string;
  title?: string | null;
  description?: string | null;
  tags?: string[];
  thumbnailAssetId?: string | null;
  thumbnailUrl?: string | null;
  publishedAt?: string | null;
};

export type GalleryListResponse = {
  items: GalleryCard[];
  limit: number;
  offset: number;
  hasMore: boolean;
};

export type ListGalleryOptions = {
  limit?: number;
  offset?: number;
};

/**
 * GET /api/v1/public/gallery — anonymous list of gallery-ready tools.
 */
export async function listGallery(
  options?: ListGalleryOptions,
): Promise<GalleryListResponse> {
  const params = new URLSearchParams();
  if (options?.limit != null) params.set("limit", String(options.limit));
  if (options?.offset != null) params.set("offset", String(options.offset));
  const qs = params.toString();
  const url = `${getApiBaseUrl()}/api/v1/public/gallery${qs ? `?${qs}` : ""}`;

  const res = await fetch(url, {
    method: "GET",
    credentials: "omit",
  });

  if (!res.ok) {
    const text = await res.text().catch(() => "");
    throw new Error(
      `List gallery failed (${res.status})${text ? `: ${text}` : ""}`,
    );
  }

  return res.json() as Promise<GalleryListResponse>;
}

/**
 * GET /api/v1/public/gallery/{publicId} — one card (404 if not gallery-ready).
 */
export async function getGalleryItem(
  publicId: string,
): Promise<GalleryCard> {
  const res = await fetch(
    `${getApiBaseUrl()}/api/v1/public/gallery/${encodeURIComponent(publicId)}`,
    {
      method: "GET",
      credentials: "omit",
    },
  );

  if (!res.ok) {
    const text = await res.text().catch(() => "");
    throw new Error(
      `Get gallery item failed (${res.status})${text ? `: ${text}` : ""}`,
    );
  }

  return res.json() as Promise<GalleryCard>;
}

File: apps/api/src/schemas/gallery.py
"""M8d — public gallery list / detail (no auth)."""

from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field


class CamelModel(BaseModel):
    model_config = ConfigDict(
        populate_by_name=True,
        serialize_by_alias=True,
        extra="ignore",
    )


class GalleryCardResponse(CamelModel):
    """
    Card payload for gallery list/detail.
    No owner, draft, code, or internal UUID.
    """

    public_id: str = Field(alias="publicId")
    title: str | None = None
    description: str | None = None
    tags: list[str] = Field(default_factory=list)
    thumbnail_asset_id: str | None = Field(
        default=None,
        alias="thumbnailAssetId",
    )
    thumbnail_url: str | None = Field(default=None, alias="thumbnailUrl")
    published_at: str | None = Field(default=None, alias="publishedAt")


class GalleryListResponse(CamelModel):
    items: list[GalleryCardResponse]
    limit: int
    offset: int
    has_more: bool = Field(alias="hasMore")
